python/pyspark/context.py

```diff
@@ -864,8 +865,21 @@ def union(self, rdds):
         first_jrdd_deserializer = rdds[0]._jrdd_deserializer
         if any(x._jrdd_deserializer != first_jrdd_deserializer for x in rdds):
             rdds = [x._reserialize() for x in rdds]
-        cls = SparkContext._jvm.org.apache.spark.api.java.JavaRDD
-        jrdds = SparkContext._gateway.new_array(cls, len(rdds))
+        gw = SparkContext._gateway
+        jvm = SparkContext._jvm
+        jrdd_cls = jvm.org.apache.spark.api.java.JavaRDD
+        jpair_jrdd_cls = jvm.org.apache.spark.api.java.JavaPairRDD
+        jdouble_jrdd_cls = jvm.org.apache.spark.api.java.JavaDoubleRDD
```

Comment:
Ideally we should do something just like:

```python
cls = JavaClass(rdds[0]._jrdd.getClass().getCanonicalName(), SparkContext._gateway)
jdstreams = SparkContext._gateway.new_array(cls, len(rdds))
```

so we forget about the signature matching in Python side but seems not working in Py4J side yet - it seems not respecting the parent and inheritance. e.g.) if we have `JavaLongRDD` extends `JavaRDD`, it doesn't work.